src/helpers/fetchAllLambdaPolicies.ts

```diff
@@ -0,0 +1,44 @@
+import { GetPolicyCommand } from '@aws-sdk/client-lambda';
+
+import { ARN, build } from '@aws-sdk/util-arn-parser';
+import { lambdaClient } from '../clients';
+import { Resource } from '../types';
+import { filterLambdaFromResources } from './filterLambdaFromResources';
+
+// Incomplete, update if needed
+export type Policy = {
+  Statement?: {
+    Condition?: {
+      ArnLike?: {
+        'AWS:SourceArn'?: string;
+      };
+    };
+  }[];
+};
+
+const fetchLambdaPolicyByArn = async (
+  arn: ARN,
+): Promise<Policy | undefined> => {
+  try {
+    const commandOutput = await lambdaClient.send(
+      new GetPolicyCommand({ FunctionName: arn.resource }),
+    );
+    const policy = JSON.parse(commandOutput.Policy ?? '') as Policy;
+
+    return policy;
+  } catch (e) {
+    return;
+  }
+};
```

Comment:
Pourquoi le try and catch ici ?

Reply:
la requête fait 404 si il n'y a pas de policy